app/upgrades_test.go

```diff
@@ -16,16 +20,47 @@ type IntegrationTestSuite struct {
 	ctx sdk.Context
 }
 
+func TestIntegrationTestSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationTestSuite))
+}
+
 func (s *IntegrationTestSuite) SetupSuite() {
 	s.app = Setup(s.T())
```

Comment:
I haven't checked the result of the github actions, but last I knew, you couldn't call `Setup` in this directory. I didn't nail down the cause but I think it had to do with some `init()` functions somewhere.

If the tests are passing, though, then I'm wrong, and it's al good (and you can resolve this comment). But if the these tests are failing, that's probably what's up with them. I'm not yet sure how best to address that problem.

Reply:
I see what you mean.  Adding:
 ```func TestIntegrationTestSuite(t *testing.T) {
	suite.Run(t, new(IntegrationTestSuite))
}```
Actually, started executing the tests in the upgrades_test.go file.  The `TestUpgradeICA` we had for a previous upgrade was not being executed. Running the test after my changes works, but you need to run it by itself.  Running all the tests in the package causes a failure.  Something to do with the InitGenesis and creating the fee_module account when setup is called.

Reply:
I mentioned this in slack, but thought I'd put it here too in case we're wondering about it in the future.

The reason the tests were failing in here when we tried to use that `Setup(s.T())` function has to do with concurrent interactions from the `app/prefix_test.go` tests. Those tests call the `SetConfig` function which (among other things) sets the bech32 HRP to either `tp` or `pb` as global variables.

What was happening is that, when running all the tests in `./app/...`:
1. Part of the `Setup` process was happening with the bech32 HRP set to `cosmos` (the default). It gets through the part where it generates a `genesis.json` file where all the addresses in it start with `cosmos1`.
2. The tests in `app/prefix_test.go` run and complete and change the HRP to `tp` in the process.
3. The `Setup` process gets to the point where it calls `InitGenesis`. Now, though, it can't decode those bech32 addresses because it's expecting them to have the HRP `tp` but they have `cosmos`. That error from `AccAddressFromBech32` is getting ignored, though, in the `BaseAccount.GetAddress()`, which instead just returns an empty address. In turn, that caused the actual error in question because that address is being provided to a `store.Set(key, value)` call as the `value`, which cannot be empty.

The fix is to put the `app/prefix_test.go` tests in a different package (e.g. `app_test`), or move them to a different directory. That way, nothing else is affected by the calls to `SetConfig`.